R/tools.R

```diff
@@ -26,28 +26,49 @@ lint_dir <- function(path) {
   lints
 }
 
+lint_path <- function(path) {
+  if (fs::is_dir(path)) {
+    lint_dir(path)
+  } else {
+    if (interactive()) {
+      message(cli::format_inline("Linting {.file {path}}"))
+    }
+    lintr::lint(filename = path)
+  }
+}
+
 #' Lint R
 #'
 #' Uses the `{lintr}` package to check all R sources in the `app` and `tests/testthat` directories
 #' for style errors.
 #'
 #' The linter rules can be adjusted in the `.lintr` file.
 #'
-#' You can set the maximum number of accepted style errors
-#' with the `legacy_max_lint_r_errors` option in `rhino.yml`.
-#' This can be useful when inheriting legacy code with multiple styling issues.
+#' You can set the maximum number of accepted style errors with the `legacy_max_lint_r_errors`
+#' option in `rhino.yml`. This can be useful when inheriting legacy code with multiple styling
+#' issues.
```

Comment:
I guess the line breaks here might have seen a bit arbitrary to you :laughing: I use what I call "semantic line breaks": I break the line after each sentence / sub-sentence. I find that it leads to smaller, easier to read diffs and less reformatting.

Compare this to breaking when you run out of space in the line: If you add or remove a couple of words, you'll often need to adjust the line breaks in the whole paragraph.

Reply:
I see. You are right I was confused on the line breaks. I will return them to the previous line breaks, and will follow your semantic line breaks.